ansible/roles/cyhy_feeds/tasks/main.yml

```diff
@@ -0,0 +1,250 @@
+    dest: /var/cyhy/scripts/cyhy-feeds/nsd_cyhy_daily_extract_public.key
+    mode: 0644
+    owner: cyhy
+    group: cyhy
+
+- name: Copy gpg private key for moe data extract
+  copy:
+    src: /tmp/ncats_lab_private_gpg.key
+    dest: /var/cyhy/scripts/cyhy-feeds/ncats_lab_private_gpg.key
+    mode: 0444
+    owner: cyhy
+    group: cyhy
+
+- name: Copy gpg trust file
+  copy:
+    src: /tmp/gpg_keys.trust
+    dest: /var/cyhy/scripts/cyhy-feeds/gpg_keys.trust
+    mode: 0644
+    owner: cyhy
+    group: cyhy
+
+- name: Create the /var/cyhy/scripts/cyhy-feeds/cyhy_extracts directory
+  file:
+    path: /var/cyhy/scripts/cyhy-feeds/cyhy_extracts
+    state: directory
+    owner: cyhy
+    group: cyhy
+
+- name: import lab public key
+  shell: "gpg2 --trustdb-name /var/cyhy/.gnupg/trustdb.gpg --import /var/cyhy/scripts/cyhy-feeds/ncats_lab_public_gpg.key"
+  become_user: cyhy
+  vars:
+    ansible_ssh_pipelining: yes
+
+- name: import nsd daily public key
+  shell: "gpg2 --trustdb-name /var/cyhy/.gnupg/trustdb.gpg --import /var/cyhy/scripts/cyhy-feeds/nsd_cyhy_daily_extract_public.key"
+  become_user: cyhy
+  vars:
+    ansible_ssh_pipelining: yes
+
+- name: import lab private key
+  shell: "gpg2 --batch --import /var/cyhy/scripts/cyhy-feeds/ncats_lab_private_gpg.key"
+  become_user: cyhy
+  vars:
+    ansible_ssh_pipelining: yes
+
+- name: Add trust
+  shell: "gpg2 --import-ownertrust /var/cyhy/scripts/cyhy-feeds/gpg_keys.trust"
+  become_user: cyhy
+  vars:
+    ansible_ssh_pipelining: yes
+
+- name: Delete nsd public key
+  local_action:
+    module: file
+    path: /tmp/nsd_cyhy_daily_extract_public.key
+    state: absent
+  become: no
+
+- name: Delete local copy config file for moe data extractor
+  local_action:
+    module: file
+    path: /tmp/cyhy-data-extract.cfg
+    state: absent
+  become: no
+
+- name: Delete local copy gpg public key for moe data extract
+  local_action:
+    module: file
+    path: /tmp/ncats_lab_public_gpg.key
+    state: absent
+  become: no
+
+- name: Delete local copy gpg public key for moe data extract
+  local_action:
+    module: file
+    path: /tmp/ncats_lab_private_gpg.key
+    state: absent
+  become: no
+
+- name: Delete gpg trust file
+  local_action:
+    module: file
+    path: /tmp/gpg_keys.trust
+    state: absent
+  become: no
+
+- name: Create /etc/cyhy/cyhy.conf with commander credentials
+  copy:
+    dest: /etc/cyhy/cyhy.conf
+    owner: cyhy
+    group: cyhy
+    mode: 0660
+    content: |
+      [DEFAULT]
+      default-section = production
+      database-uri = mongodb://localhost:27017/
+      report-key =
+
+      [production]
+      database-uri = mongodb://commander:{{ mongo_users.other_users | selectattr("user", "eq", "commander") | map(attribute="password") | first }}@database1:27017/cyhy
+      database-name = cyhy
+
+#
+# Run nightly at 0815 (UTC) as cyhy
+#
+- name: Set up nightly cron job to sync NSD data for moe extract
+  cron:
+    name: Nightly cyhy extract
+    hour: 08
```

Comment:
Just checking to make sure you're aware that the machines are on UTC time.  So this will run at 0415 EST or 0315 EDT.

Reply:
It doesn't matter what time it runs as long as it is daily.